### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by eli-schwartz**
Source code, e.g. #include, belongs in the source code, not the build system. Build systems are for defining include paths, not the includes themselves.

Drop the KeySym keyword since it seems to be a useless alias for uint32_t which might as well be used directly instead of indirectly using #define everywhere.

Also restore the desktop file, since desktop files do not go stale. llpp has not stopped accepting the CLI semantic `llpp /path/to/file.pdf`.

### CODE REVIEW

**Comment by moosotc** [comment]
Eli Schwartz <notifications@github.com> writes:

> Source code, e.g. #include, belongs in the source code, not the build
> system. Build systems are for defining include paths, not the includes
> themselves.
>

That's both:
over-simplifcation and over-generalization

Furthermore build system (i.e. build.bash) is not changed by the patch
in the slightest.

> Drop the KeySym keyword since it seems to be a useless alias for
> uint32_t which might as well be used directly instead of indirectly
> using #define everywhere.

Think i have an idea how to do it differently (and perhaps better), will
try to see if i can implment this idea of mine (sooner rather than later
i hope)

>
> Also restore the desktop file, since desktop files do not go
> stale. llpp has not stopped accepting the CLI semantic `llpp
> /path/to/file.pdf`.  You can view, comment on, or merge this pull
> request online at:
>

Even at a VERY quick glance the desktop contains two instance of
image/png, i have no idea what else is wrong with it.

FWIW i find xdg-open's shell parts erregriously bad and do not want to
have anything to do with it (even if only by association) sorry :(

So basically "no" on the desktop parts, i will not apply it.

>   https://github.com/moosotc/llpp/pull/85
>
> -- Commit Summary --
>
>   * Do not use CPPFLAGS to add headers to source files
>   * Revert "This likely became stale"
>
> -- File Changes --
>
>     M link.c (4)
>     A misc/llpp.desktop (14)
>     M wsi/x11/keysym2ucs.c (4)
>
> -- Patch Links --
>
> https://github.com/moosotc/llpp/pull/85.patch
> https://github.com/moosotc/llpp/pull/85.diff

-- 
mailto:moosotc@gmail.com

**Comment by eli-schwartz** [comment]
Very well, I've deleted the desktop file restoration.

I didn't try updating the build.bash because I don't understand how it works, and because it currently doesn't build in the first place (so I'm kind of afraid to touch it). I use a custom Makefile instead, which just needed updating for new files that you've added recently.

**Comment by eli-schwartz** [comment]
> Even at a VERY quick glance the desktop contains two instance of image/png, i have no idea what else is wrong with it.

Looks like that was added in 4c3723415534f66e9b9e7f0a1018d6e5d409536c but I don't know why...

**Comment by moosotc** [comment]
Eli Schwartz <notifications@github.com> writes:

> Very well, I've deleted the desktop file restoration.
>
> I didn't try updating the build.bash because I don't understand how it
> works, and because it currently doesn't build in the first place (so
> I'm kind of afraid to touch it). I use a custom Makefile instead,
> which just needed updating for new files that you've added recently.

I'm very interested to know why/how it doesn't work for you, it works
fine here on this arch box, it worked fine when i tried it two days ago
on an Ubuntu machine that belongs to a friend, it also works just fine
on (at least) two macOS machines...

-- 
mailto:moosotc@gmail.com

**Comment by moosotc** [comment]
Eli Schwartz <notifications@github.com> writes:

>> Even at a VERY quick glance the desktop contains two instance of
> image/png, i have no idea what else is wrong with it.
>
> Looks like that was added in 4c3723415534f66e9b9e7f0a1018d6e5d409536c
> but I don't know why...

I should probably invent a time machine and go back in time to
interrogate moron who put it there. Sigh. If it shows anything is that
i'm incapable of maintaining it (nor, for that matter, most of the stuff
i personally never use :( )

-- 
mailto:moosotc@gmail.com

**Comment by eli-schwartz** [comment]
> I'm very interested to know why/how it doesn't work for you, it works fine here on this arch box, it worked fine when i tried it two days ago on an Ubuntu machine that belongs to a friend, it also works just fine on (at least) two macOS machines...

I was getting the "inconsistent assumptions over interface Wsi" message mentioned (I think this is the same issue?) in a2fff17750b95146ea5a58889e3f30b95413f763, initially due to an (erroneously declared) dependency cycle in my Makefile, except I got the same issue when I tried running build.bash... I didn't investigate further, because I tracked down the loop in the Makefile.

**Comment by moosotc** [comment]
Eli Schwartz <notifications@github.com> writes:

>> I'm very interested to know why/how it doesn't work for you, it
> works fine here on this arch box, it worked fine when i tried it two
> days ago on an Ubuntu machine that belongs to a friend, it also works
> just fine on (at least) two macOS machines...
>
> I was getting the "inconsistent assumptions over interface Wsi"
> message mentioned (I think this is the same issue?) in
> a2fff17750b95146ea5a58889e3f30b95413f763, initially due to an
> (erroneously declared) dependency cycle in my Makefile, except I got
> the same issue when I tried running build.bash... I didn't investigate
> further, because I tracked down the loop in the Makefile.

Can you do a fresh checkout of llpp (in /tmp) and then follow the
instructions in BUILDING to the letter? (The inconsistent thing should
never happen with build.bash... ever)

-- 
mailto:moosotc@gmail.com
